internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -0,0 +1,162 @@
+package resourceRequestOperator
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	"github.com/liqotech/liqo/pkg/discovery"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/util/slice"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"time"
+)
+
+// ResourceRequestReconciler reconciles a ResourceRequest object
+type ResourceRequestReconciler struct {
+	client.Client
+	Log       logr.Logger
+	Scheme    *runtime.Scheme
+	ClusterId string
+}
+
+type ResourceToOffer struct {
+	Offers corev1.ResourceList
+}
+
+var resources ResourceToOffer
+
+const (
+	finalizerName = "discovery.liqo.io.resourceRequest.finalizer"
+	offerPrefix   = "resourceoffer-"
+)
+
+// +kubebuilder:rbac:groups=discovery.liqo.io,resources=resourceRequests,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=discovery.liqo.io,resources=resourceRequests/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=sharing.liqo.io,resources=resourceOffers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=sharing.liqo.io,resources=resourceOffers/status,verbs=get;update;patch
+
+func (r *ResourceRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	var resourceRequest discoveryv1alpha1.ResourceRequest
+	err := r.Get(ctx, req.NamespacedName, &resourceRequest)
+	if err != nil {
```

Comment:
Why do you differentiate the errors? I would replace the whole if statement with a simple `if err != nil` without distinguishing between the error codes.